node/pkg/tss/marshal.go

```diff
@@ -90,56 +45,52 @@ func (s *GuardianStorage) load(storagePath string) error {
 		return err
 	}
 
-	s.signingKey = unmarshalEcdsaSecretKey(s.SecretKey)
+	// tlsPrivateKey to signingKey.
+	s.signingKey, err = internal.PemToPrivateKey(s.PrivateKey)
+	if err != nil {
+		return fmt.Errorf("error parsing tls private key: %v", err)
+	}
 
-	pk, err := unmarshalEcdsaPublickey(tss.S256(), s.Self.Key)
+	pk, err := internal.PemToPublicKey(s.Self.Key)
 	if err != nil {
 		return err
 	}
 
 	if !s.signingKey.PublicKey.Equal(pk) {
-		return fmt.Errorf("signing key does not match the public key stored as Self partyId")
+		return fmt.Errorf("signing key does not match the public key stored in Self.Key")
 	}
 
-	if !tss.S256().IsOnCurve(pk.X, pk.Y) {
+	// just to be safe.
+	if !s.signingKey.Curve.IsOnCurve(pk.X, pk.Y) {
```

Comment:
What does this check achieve? is it just to make sure nothing got messed up in the formatting or is it for security?

For security, I'm not sure being on the Curve is enough. It might also need to be in the right subgroup.

Reply:
Task: 
check the proper way to ensure a public key, is on the correct subgroup. 
`g^{order of the specific sub group of p256} = 1 is safe public key.`